Week3/app.js

```diff
@@ -0,0 +1,144 @@
+    }
+}
+
+// More JavaScript 
+// 1.Create a function that takes 3 arguments and returns the sum of the these arguments.
+function sumArg(arg1, arg2, arg3) {
+    // let sum = arg1 + arg2 + arg3;
+    // return sum;
+    return arg1 + arg2 + arg3;
+}
+console.log(sumArg(1, 2, 3));
+
+// 2.Create a function named colorCar that receives a color, and prints out, 'a red car' for example.
+function colorCar(color) {
+    let carColor = "a " + color + " car";
+    return carColor;
+}
+console.log(colorCar("red"));
+
+// 3.Create an object and a function that takes the object as a parameter and prints out all of its properties and values.
+const testObj = {
+    s: "Name",
+    d: "Age",
+    f: "DOB"
+};
+function proObj(obj) {
+    for (let [key, value] of Object.entries(obj)) {
+        console.log(`${key}: ${value}`);
+    }
+}
+proObj(testObj);
+
+// 4.Create a function named vehicleType that receives a color, and a code, 1 for car, 2 for motorbike. And prints 'a blue motorbike' for example when called as vehicleType("blue", 2)
+let codeVehicle = {
+    1: "car",
+    2: 'motorbike'
+};
+function vehicleType(color, code) {
+    console.log("a " + color + " " + codeVehicle[code]);
+}
+vehicleType("blue", 2);
+
+// 5.Can you write the following without the if statement, but with just as a single line with console.log(...);?
+let h = (3 === 3) ? "yes" : "no";
+console.log(h);
+
+// 6.Create a function called vehicle, like before, but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+function vehicle(color, code, age) {
+    if (age === 1) {
+        console.log("a " + color + " " + "new " + codeVehicle[code]);
+    } else{
+        console.log("a " + color + " " + "used " + codeVehicle[code]);
+    }
+}
+vehicle("blue", 1, 5);
+
+// 7.Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.
+let listVehicles = ["motorbike", "caravan", "bike"];
+// 8.How do you get the third element from that list?
+console.log(listVehicles[2]);
+
+// 9.Change the function vehicle to use the list of question 7. So that vehicle("green", 3, 1) prints "a green new bike".
+function newVehicle(color, code, age) {
+    if (age === 1) {
+        console.log("a " + color + " " + "new " + listVehicles[code - 1]);
+    } else{
+        console.log("a " + color + " " + "used " + listVehicles[code - 1]);
+    }
+}
+newVehicle("green", 3, 1);
+
+// 10.Use the list of vehicles to write an advertisement. So that it prints something like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.". (Hint: use a for loop.)
+let advertisement = "Amazing Joe's Garage, we service ";
+for (let q = 0; q < listVehicles.length; q++){
+    if (q < (listVehicles.length-1)){
+        advertisement = advertisement +  ", " + listVehicles[q];
+    } else if (q === (listVehicles.length - 1)){
+        advertisement = advertisement + " and " + listVehicles[q] + ".";
+    }
+}
+console.log(advertisement);
```

Comment:
Hi Huseyin - there seems to be a problem here. The expected output was: `Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.` but what I see happening is: `Amazing Joe's Garage, we service , motorbike, caravan and bike.`. Could you please check?

Reply:
okey,  i'll check it and improve.